src/libs/ReportActionsUtils.ts

```diff
@@ -372,7 +373,25 @@ function replaceBaseURL(reportAction: ReportAction): ReportAction {
     if (!updatedReportAction.message) {
         return updatedReportAction;
     }
-    updatedReportAction.message[0].html = reportAction.message[0].html?.replace('%baseURL', environmentURL);
+    const reportID = reportAction.originalMessage.reportID;
+    const roomName = allReports ? allReports[`${reportID}`]?.reportName : '';
```

Comment:
Interesting case. I am not sure what solution we have for this case then.

Reply:
The RCA for this issue is that we save the `roomName` in the DB for policy change and Invite action. Now, if the roomName gets updated, then we don't update the roomName in those actions. As a result, we have stale room name being rendered.

I want to take a step back and re-think the solution here. I can update the code in the backend that will always send the latest roomName since we have the reportID with us. So that will leave us with no change required in the frontend, right? In case of invite/remove action, I can also send the updated list of policy change actions to the #admin room.

Does that sound good?

Reply:
> But with that we will need to update all reportAction that is related from the room, it might take the API is delayed.

We can just update the policy change logs actions `POLICYCHANGELOG_INVITETOROOM`?

Reply:
> In this case do you think we can use the old room name from originalMesssage.roomName.

Did you test this? Is the linked chat room loaded in onyx? If it is working then I think it is acceptable.

Reply:
Hey, do we have any way forward here? I think for now we can just stick with updating the roomName in the backend described here: https://github.com/Expensify/App/pull/30981#discussion_r1396827440

Reply:
Sorry, I would need to discuss this with @jasperhuangg, who worked on this feature, to figure out the next steps here, sorry for the delay.